### ISSUE TITLE
Issue when the network is composed of one node

### ISSUE THREAD

**Issue by MohamedLEGH** [makgyver/gossipy#9]
I know that it doesn't make sense to create a network with one node but I did it to compare results with a centralized solution.
This is the code used (adapted from `main_ormandi.py`:
```python
from gossipy.gossipy import set_seed
from gossipy.gossipy.core import AntiEntropyProtocol, CreateModelMode, StaticP2PNetwork, UniformDelay
from gossipy.gossipy.node import GossipNode
from gossipy.gossipy.model.handler import PegasosHandler
from gossipy.gossipy.model.nn import AdaLine
from gossipy.gossipy.data import load_classification_dataset, DataDispatcher
from gossipy.gossipy.data.handler import ClassificationDataHandler
from gossipy.gossipy.simul import GossipSimulator, SimulationReport
from gossipy.gossipy.utils import plot_evaluation

set_seed(42)
X, y = load_classification_dataset("spambase", as_tensor=True)
y = 2*y - 1 #convert 0/1 labels to -1/1

data_handler = ClassificationDataHandler(X, y, test_size=.1)
data_dispatcher = DataDispatcher(data_handler, 1, eval_on_user=False, auto_assign=True)
topology = StaticP2PNetwork(1, None)
model_handler = PegasosHandler(net=AdaLine(data_handler.size(1)),
                               learning_rate=.01,
                               create_model_mode=CreateModelMode.MERGE_UPDATE)

# For loop to repeat the simulation
nodes = GossipNode.generate(data_dispatcher=data_dispatcher,
                            p2p_net=topology,
                            model_proto=model_handler,
                            round_len=100,
                            sync=False)

simulator = GossipSimulator(
    nodes=nodes,
    data_dispatcher=data_dispatcher,
    delta=100,
    protocol=AntiEntropyProtocol.PUSH,
    delay=UniformDelay(0,10),
    online_prob=.2, #Approximates the average online rate of the STUNner's smartphone traces
    drop_prob=.1, #Simulate the possibility of message dropping,
    sampling_eval=.1
)

report = SimulationReport()
simulator.add_receiver(report)
simulator.init_nodes(seed=42)
simulator.start(n_rounds=100)

plot_evaluation([[ev for _, ev in report.get_evaluation(False)]], "Overall test results")
```

And the results I get back are the following :
```bash
# INFO     # Sent messages: 95                                                                                   simul.py:248
# INFO     # Failed messages: 79                                                                                 simul.py:249
# INFO     Total size: 5415                                                                                      simul.py:250
# INFO     accuracy: 0.62                                                                                        utils.py:181
# INFO     precision: 0.62                                                                                       utils.py:181
# INFO     recall: 0.62                                                                                          utils.py:181
# INFO     f1_score: 0.61                                                                                        utils.py:181
# INFO     auc: 0.66                                                                                             utils.py:181
```

 In the algorithm from the paper `ormandi2013`, a node wait for the reception of a model to update the model on the data, and thus the model never learn. This is a bit sad but expected. The part that is very strange for me is the number of "failed "message".
I would have suppose that all message would failed since there is no other nodes in the network.
What do you think ? Is this expected or a bug ?

**Comment by makgyver** [comment]
Hi @MohamedLEGH, I just had a quick look to the `GossipSimulator.start` method and maybe there is a miscounting at line 399. I think the line `self.notify_message(False, msg)` should be moved inside the if statement (then branch). I will check again tomorrow and if it is confirmed I fix it.

**Comment by makgyver** [comment]
Hi @MohamedLEGH, I found the issue, and it was not the one I mentioned in the previous message. The problem was in the `StaticP2PNetwork` and in particular the method `get_peers(self, idx)` that in case of a clique topology returned the list of indices of the clients **including** `idx`! So the counting of the sent messages was correct because it was trying to send the messages to itself and some of them failed due to the `drop_prob` and the `online_prob`. I just fixed it - 9bb203e.
